File: RegexTester/LargestComparer.vb
Imports System
Imports System.Collections.Generic
Imports System.Text.RegularExpressions

Namespace RegexTester
    Public Class LargestComparer
        Implements IComparer(Of Match)
        ' Methods
        Public Function [Compare](ByVal x As Match, ByVal y As Match) As Integer Implements IComparer(Of Match).Compare
            Return y.Length.CompareTo(x.Length)
        End Function

    End Class
End Namespace

File: RegexTester/LargestComparer.cs
namespace RegexTester
{
    using System;
    using System.Collections.Generic;
    using System.Text.RegularExpressions;

    public class LargestComparer : IComparer<Match>
    {
        public int Compare(Match x, Match y)
        {
            return y.Length.CompareTo(x.Length);
        }
    }
}

